File: server/model/base/request/logReg.go
package request

type Login struct {
	Username  string `json:"username" validate:"required"`  // 用户名
	Password  string `json:"password" validate:"required"`  // 密码
	Captcha   string `json:"captcha" validate:"required"`   // 验证码
	CaptchaId string `json:"captchaId" validate:"required"` // 验证码ID
}

type CaptchaResponse struct {
	CaptchaId     string `json:"captchaId"`
	PicPath       string `json:"picPath"`
	CaptchaLength int    `json:"captchaLength"`
}

File: web/src/api/base/logReg.ts
import { request } from "@/utils/service"

export interface LoginRequestData {
  /** admin 或 editor */
  username: string
  /** 密码 */
  password: string
  /** 验证码 */
  captcha: string
  captchaId: string
}

type LoginCodeResponseData = ApiResponseData<{ picPath: string; captchaId: string }>
type LoginResponseData = ApiResponseData<{ token: string }>

// 获取验证码
export const captcha = () => {
  return request<LoginCodeResponseData>({
    url: "/logReg/captcha",
    method: "post"
  })
}

/** 登录并返回 Token */
export function loginApi(data: LoginRequestData) {
  return request<LoginResponseData>({
    url: "/logReg/login",
    method: "post",
    data
  })
}

// 登出
export function logoutApi() {
  return request<ApiResponseData<null>>({
    url: "/logReg/logout",
    method: "post",
    data: {}
  })
}
